docs/DeveloperGuide.md

```diff
@@ -2,33 +2,249 @@
+                might be useful in further development of implementation of the use of the `ScheduleManager`.
+                Searching for a date can also be done in O(logn) similarly to a `HashMap`.
+    - Cons: Storing the dates that have past and also storing of many dates might be heavy in
+                terms of space.
+- Use of a `HashMap` to store the date as keys and `ArrayList` as values
+    - Pros: Allows tracking of tasks with respect to the dates as keys.
+    - Cons: Does Not allow iteration in order as when desired as compared to a `TreeMap`.
+- Creation of a new class which contains a schedule for each day, and using a `TreeSet`.
+      to contain this new class.
+    - Pros: Making use of more oop design.
+    - Cons: Might be tricky to implement and may need more resources.
+    
+### Storage
+The application will save all of the user inputs by using a `Storage` class, which is designed as
+singleton.
+
+![](Images/3.5Stroage.JPG)
+
+It exports the data in the application and converts it into the form of a txt file. This allows users
+to retain their saved data after exiting the app. More elaboration of how this will be done will be shown later, 
+where a sequence diagram will show how this process works.
+
+## 4. Implementation
+
+This section describes some noteworthy details on how certain features are
+implemented.
+
+### Add Feature
+
+The add implementation allows the user to add a `Lesson`, `Event` or `Deadline`. This is facilitated
+by implementing the ScheduleManager and ModuleManager, which stores all of the `Task` in
+them. The `ScheduleManager` helps to save these tasks into the dates according to the date of
+the `Task` as stated by the User. The `ModuleManager` will also add the `Task` into the respective module
+that it belongs to. Below is a sequence diagram which shows how adding of a
+`Lesson` works. Adding of a Event or a Deadline is also in a similar fashion.
+
+![](Images/4.1Add.JPG)
+
+### Display Feature
+
+![](Images/4.2Display.JPG)
+
+For display method, there are 5 possible scenarios that could include parameters such
+as `Module` code, date & range of date.
+As seen from the diagram, the 5 Scenarios are:
+1. Display today Schedule
+2. Display all information in a specific module
+3. Display all task in a specific module on a specific date
+4. Display all task within a range of dates
+5. Display all task on a specific date
+
+Example: flow for Displaying today schedule:
+1. The user will run the app and type in “display”, which will then be received by the `Ui` and
+it returns the input as a String fullCommand
+2. The String fullCommand will then be parsed by a `Parser` to return an `DisplayCommand`
+object.
+3. The DisplayCommand object will then be executed to display today’s Schedule using the
+`ScheduleManager.displayToday()` method
+
+### Storage Feature
+
+1. Set up the `Storage` from local disk 
+
+![](Images/4.3Storage1.JPG)
```

Comment:
![4 3Storage1](https://user-images.githubusercontent.com/49282734/97662078-0926bf80-1ab1-11eb-9e28-58c502f2858d.jpeg)

When Storage calls its own method, I believe it should follow the self-invocation diagram convention instead? Also, the <<class>> might not be necessary here, since this is intended to be a sequence diagram and not an object/class diagram.

Reply:
storage(Directorypath) line draw lower